### ISSUE TITLE
Allow OrmLiteConfig.StringFilter to fire on null strings

### ISSUE THREAD

**Issue by robertmiles3** [ServiceStack/Issues#306]
In case one wanted to have `OrmLiteConfig.StringFilter` deal with a null string coming back from the DB (like setting it to `string.Empty` instead), it would be nice to allow the `StringFilter` to be called on the null string. However, currently it doesn't because the `as` conversion returns null and therefore skips it. A downside is that I could see this being a breaking change for users who are currently not handling null possibilities within their `StringFilter` implementations.

Seems like a design-related question whether it should be allowed, so thought I'd ask. If it stays as-is, can you think of another option to have a `StringFilter`-esque operation on null strings?

**Comment by mythz** [comment]
Wont be called in StringFilter but I've just added a new generic `OrmLiteConfig.OnDbNullFilter` that can be used for any null value instead in this commit: https://github.com/ServiceStack/ServiceStack.OrmLite/commit/e8573202a5781522bf236992d109449370096412

Where you can return "NULL" for null strings with:

```csharp
OrmLiteConfig.OnDbNullFilter = fieldDef => 
    fieldDef.FieldType == typeof(string)
        ? "NULL"
        : null;
```

This change is available from v4.0.41+ that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

In future please reserve this Issues site for reproducible bugs/issues, (i.e. runnable code with failing tests). You can use https://forums.servicestack.net/ for general advice, design or feature discussion, etc or [UserVoice](servicestack.uservoice.com/forums/176786-feature-requests) for feature requests.

**Comment by robertmiles3** [comment]
Thanks Demis. Sounds good. Sorry for the wrong spot. Hope all is well!
